src/binder/bind/bind_copy.cpp

```diff
@@ -36,21 +39,60 @@ static void validateCopyNpyKeyword(
     CopyDescription::FileType expectedType, CopyDescription::FileType actualType) {
     if (expectedType == CopyDescription::FileType::UNKNOWN &&
         actualType == CopyDescription::FileType::NPY) {
-        throw BinderException("Please use COPY FROM BY COLUMN statement for copying npy files.");
+        throw BinderException(ExceptionMessage::validateCopyNPYByColumnException());
     }
     if (expectedType == CopyDescription::FileType::NPY && actualType != expectedType) {
-        throw BinderException("Please use COPY FROM statement for copying csv and parquet files.");
+        throw BinderException(ExceptionMessage::validateCopyCSVParquetByColumnException());
     }
 }
 
-static void validateCopyNpyFilesMatchSchema(uint32_t numFiles, catalog::TableSchema* schema) {
+static void validateCopyNpyFilesMatchSchema(uint32_t numFiles, TableSchema* schema) {
     if (schema->properties.size() != numFiles) {
         throw BinderException(StringUtils::string_format(
             "Number of npy files is not equal to number of properties in table {}.",
             schema->tableName));
     }
 }
 
+static void validateCopyNpyNotForRelTables(TableSchema* schema) {
+    if (schema->tableType == TableType::REL) {
+        throw BinderException(
+            ExceptionMessage::validateCopyNpyNotForRelTablesException(schema->tableName));
+    }
+}
+
+expression_vector Binder::bindColumnExpressions(TableSchema* tableSchema) {
+    expression_vector columnExpressions;
+    if (tableSchema->tableType == TableType::REL) {
```

Comment:
Can we refactor this to a function inside the catalog?
E.g. getPropertiesToCopy() 
And let nodeTableSchema and relTableSchema implement their own version of  getPropertiesToCopy() 
I think this will be used in multiple places.

Reply:
Let's refactor when we are using them in multiple places. An alternative solution could be we treat FROM TO as a regular property

Reply:
> Can we refactor this to a function inside the catalog? E.g. getPropertiesToCopy() And let nodeTableSchema and relTableSchema implement their own version of getPropertiesToCopy() I think this will be used in multiple places.

I don't want to complicate TableSchema with this logic. This is more specific to COPY and READER.